docs/dev/dev_adr.md

```diff
@@ -137,3 +137,35 @@ The API view, under the path `config-postprocessing`, uses custom permissions, n
 It was decided to restrict the usage of Jinja filters to only the ones related to getting Nautobot secrets values (defined here), plus the `encrypt_type5` and `encrypt_type7` filters from Netutils. Remember that this function is not defined to replace the regular Jinja rendering done for creating the Intended configuration, only to add secrets information on the fly. This avoids undesired behavior on this synchronous operation.
 
 This function performs an additional permission validation, to check if the requesting user has permissions to view the `SecretsGroup` requested.
+
+## Configuration Provisioning
+
+This feature is about updating the configuration state of a network device. It requires a proper configuration artifact, that will come from an Intended Configuration and the [Post Processing](#configuration-postprocessing) manipulation.
+
+We can identify two types of configuration provisioning:
+
+- Full Configuration or Configuration Replace: this implies that the whole configuration store in the target device is replaced by a new one. This requires being able to render exactly the whole expected configuration.
+- Partial Configuration or Configuration Merge: this provisioning mode only targets a specific subset of configuration features, for instance only NTP or DNS.
+
+Special considerations to be taken into account to create valid configuration artifacts, that have a direct impact on the required post-processing manipulations:
+
+- Full Configuration
+  - Some devices require a configuration snippet that is internally created from CLI input, and can't be recreated externally. So, the configuration artifact would need to account for it.
+- Partial Configuration
+  - Understanding the missing or exceeding configuration doesn't always match to adding or removing, respectively, commands in the proper configuration section. Some hierarchical analysis to understand the proper remediation should be considered
+
+Desired features:
+
+- Option to rollback configuration if activation (aka "commit") fails, with timeout
+- Offer a dry-run operation, to understand the actual changes
+- Accept an approval step to push the configuration to the device
+- Same experience via UI and API, as a Nautobot Job
+- Fail fast, if the device is not reachable
+- No secrets should be leaked (stored) during the process. If a config change must be approved to deploy, a hashing mechanism would be preferred.
+- It should scale to multiple devices
+- Eventually, it should allow hooking to pre/post validation (when plugin available)
```

Comment:
I think about as well:
* Interaction with the configuration that would be pushed (e.g. select the configuration to be pushed and/or allow for user to paste in configuration that they want to push) 
* Providing modularity for how the configuration to push is generated, e.g. hier_config vs missing config only